File: blog-admin/blog_admin/blog_admin/models/social_link.py
from django.db import models


class SocialLink(models.Model):
    platform = models.CharField(max_length=50)
    url = models.CharField(max_length=500)
    label = models.CharField(max_length=100, blank=True, null=True)
    icon = models.CharField(max_length=255, blank=True, null=True)
    sort_order = models.IntegerField(default=0)
    is_active = models.BooleanField(default=True)
    created_at = models.DateTimeField(auto_now_add=True)

    class Meta:
        managed = False
        db_table = 'social_links'
        ordering = ['sort_order']

    def __str__(self):
        return f"{self.platform}: {self.label or self.url}"

File: blog-api/src/test/scala/su/wps/blog/repositories/models/SocialLink.scala
package su.wps.blog.repositories.models

import cats.syntax.functor.*
import doobie.*
import doobie.generic.auto.*
import su.wps.blog.instances.time.*
import su.wps.blog.tools.types.*

import java.time.ZonedDateTime

final case class SocialLink(
  id: PosInt,
  platform: Varchar[W.`50`.T],
  url: Varchar[W.`500`.T],
  label: Option[Varchar[W.`100`.T]],
  icon: Option[Varchar[W.`255`.T]],
  sortOrder: Int,
  isActive: Boolean,
  createdAt: ZonedDateTime
)

object SocialLink {

  object sql {
    def insert(socialLink: SocialLink): ConnectionIO[Unit] = {
      val sql =
        """
          |INSERT INTO social_links (
          | id,
          | platform,
          | url,
          | label,
          | icon,
          | sort_order,
          | is_active,
          | created_at
          |) VALUES (?, ?, ?, ?, ?, ?, ?, ?)
          |""".stripMargin
      Update[SocialLink](sql).run(socialLink).void
    }
  }
}
